src/ansys/aedt/core/application/design.py

```diff
@@ -739,9 +741,12 @@ def project_name(self):
 
         >>> oProject.GetName
         """
+        if self._project_name:
+            return self._project_name
         if self.oproject:
             try:
                 self._project_name = self.oproject.GetName()
+                self._project_path = None
```

Comment:
Why are you resetting `self._project_path` here ? This applies to other properties changes.

Reply:
because if oproject changes the cached value has to be changed as well.
We may think to revert the three cached objects I've added into this PR. I mainly did to reduce the number of API calls